core/lib/dal/Cargo.toml

```diff
@@ -21,6 +21,8 @@ zksync_health_check = { path = "../health_check" }
 zksync_consensus_roles = { version = "0.1.0", git = "https://github.com/matter-labs/era-consensus.git", rev = "97d139969476a004c50f8b4a31ece748e5bee14e" }
 zksync_consensus_storage = { version = "0.1.0", git = "https://github.com/matter-labs/era-consensus.git", rev = "97d139969476a004c50f8b4a31ece748e5bee14e" }
 zksync_protobuf = { version = "0.1.0", git = "https://github.com/matter-labs/era-consensus.git", rev = "97d139969476a004c50f8b4a31ece748e5bee14e" }
+zksync_db_connection = { path = "../db_connection" }
+zksync_prover_dal = { path = "../../../prover/prover_dal" }
```

Comment:
It sounds quite counter-intuitive to depend on a local crate outside the workspace. IMO, it'd make sense to temporarily place `prover_dal` in the main workspace and place it in `core/lib` directory, and move it to a proper directory / workspace in a follow-up PR.